app/src/main/java/ch/hikemate/app/model/authentication/FirebaseAuthRepository.kt

```diff
@@ -218,7 +218,11 @@ class FirebaseAuthRepository : AuthRepository {
       onSuccess: () -> Unit,
       onFailure: () -> Unit
   ) {
-    val email = user.email ?: throw Exception("User email is null")
+    if (user.email == null) {
+      onFailure()
+      return
+    }
+    val email = user.email!!
```

Comment:
Suggestion: What about this piece of code? This should have the same functionality, but will remove the null assertion, which is not a good practice when it can be avoided:
```suggestion
val email = user.email ?: return onFailure()
```

Reply:
Yes, it's cleaner!